flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -675,18 +722,23 @@ private <T> WatermarkGaugeExposingOutput<StreamRecord<T>> createOutputCollector(
             WatermarkGaugeExposingOutput<StreamRecord<T>> output =
                     createOperatorChain(
                             containingTask,
+                            operatorConfig,
                             chainedOpConfig,
                             chainedConfigs,
                             userCodeClassloader,
                             recordWriterOutputs,
                             allOperatorWrappers,
                             outputEdge.getOutputTag(),
-                            mailboxExecutorFactory);
+                            mailboxExecutorFactory,
+                            shouldAddMetric);
             allOutputs.add(output);
+            shouldAddMetric = false;
```

Comment:
I'd suggest to add an inline comment explaining that when there're multiple downstream operators, only one of them should add the metrics.

Reply:
Thanks for the suggestion. I have updated the PR to explain the it.